react/src/hooks/useGlobalLocalStorageState.tsx

```diff
@@ -0,0 +1,31 @@
+import { useLocalStorageState } from 'ahooks';
+import { IFuncUpdater } from 'ahooks/lib/createUseStorageState';
+import { useEffect } from 'react';
+
+export const useGlobalLocalStorageState = <T extends {}>(
+  key: string,
+  defaultValue: T,
+): [T | undefined, (value?: T | IFuncUpdater<T> | undefined) => void] => {
+  const [storageValue, setStorageValue] = useLocalStorageState<T>(key, {
+    defaultValue: defaultValue,
+  });
+
+  useEffect(() => {
+    const handler = (event: any) => {
+      setStorageValue(event.detail);
+    };
+    document.addEventListener(key, handler);
+    return () => document.removeEventListener(key, handler);
+  }, [key, setStorageValue]);
+
+  return [
+    storageValue,
+    (value) => {
+      document.dispatchEvent(
+        new CustomEvent(key, {
```

Comment:
Please add a prefix to the event name.
```suggestion
        new CustomEvent('useLocalStorageGlobalState:'+key, {
```